### PR TITLE
feat: add options for returning record data and raw responses/requests in execute_stream_test_read

### PR DESCRIPTION
**Pull request by aaronsteers**
# Simplify execute_stream_test_read to minimal implementation

## Summary

This PR dramatically simplifies the `execute_stream_test_read` function from 172 lines to ~60 lines (66% reduction) by removing complex conditional logic and helper functions. The function now always returns both record data and raw request/response metadata with secrets filtered, eliminating the need for `include_records` and `include_raw_response_data` parameters.

**Key Changes:**
- **Removed parameters**: `include_records` and `include_raw_response_data` 
- **Deleted helper functions**: `_without_records` and `_without_raw_requests_and_responses`
- **Simplified logic**: Always extract and return both records and slices data
- **Updated tests**: Removed mocked unit tests, kept integration tests
- **Maintained security**: Still applies `filter_config_secrets` to slices

## Review & Testing Checklist for Human

**⚠️ High Priority Items (5 items):**

- [ ] **Verify backward compatibility**: Check if any existing code calls `execute_stream_test_read` with the removed parameters (`include_records`, `include_raw_response_data`) 
- [ ] **Test edge cases**: Try various manifest types, error scenarios, and malformed inputs beyond the Rick and Morty example
- [ ] **Verify security**: Confirm that `filter_config_secrets` properly sanitizes all sensitive data in the slices output
- [ ] **Performance testing**: Check if always extracting both records and slices impacts memory usage or response times for large datasets
- [ ] **Real-world validation**: Test with actual production connector manifests to ensure the simplified logic handles all real-world scenarios

**Recommended Test Plan:**
1. Run the function with various connector manifests (not just Rick and Morty)
2. Test error scenarios (invalid manifests, network failures, malformed responses)
3. Verify no secrets appear in the returned slices data
4. Check performance with large record sets (test with `max_records=1000`)

---

### Diagram

```mermaid
%%{ init : { "theme" : "default" }}%%
graph TD
    Caller["Function Callers"]
    TestRead["execute_stream_test_read<br/>(connector_builder.py)"]:::major-edit
    HelperFuncs["_without_records<br/>_without_raw_requests_and_responses<br/>(DELETED)"]:::major-edit
    StreamResult["StreamTestResult<br/>(model)"]:::minor-edit
    CDK["Airbyte CDK<br/>(read_stream)"]:::context
    Tests["tests/__init__.py<br/>(mocked tests DELETED)"]:::major-edit
    IntegrationTests["tests/test_integration.py<br/>(updated assertions)"]:::minor-edit
    
    Caller -->|"calls (params removed)"| TestRead
    TestRead -->|"DELETED"| HelperFuncs
    TestRead -->|"always returns both<br/>records + slices"| StreamResult
    TestRead -->|"calls"| CDK
    TestRead -->|"applies filter_config_secrets"| StreamResult
    Tests -->|"validates"| TestRead
    IntegrationTests -->|"validates"| TestRead
    
    subgraph Legend
        L1["Major Edit"]:::major-edit
        L2["Minor Edit"]:::minor-edit  
        L3["Context/No Edit"]:::context
    end
    
    classDef major-edit fill:#90EE90
    classDef minor-edit fill:#87CEEB
    classDef context fill:#FFFFFF
```

### Notes

- **Breaking change**: This removes function parameters, so any existing callers will need updates
- **Simplified approach**: The "all or nothing" approach eliminates complex conditional logic but changes the function contract
- **CI passing**: All 63 tests pass including pytest on Python 3.10/3.11, but this doesn't guarantee backward compatibility
- **Requested by**: AJ Steers (@aaronsteers)
- **Devin session**: https://app.devin.ai/sessions/e31e9b3aeee54c569beec1aafa4e25cd

The main risk is that this is a breaking change disguised as a simplification. While the code is cleaner and the integration test passes, we need to verify no existing code depends on the removed functionality.

### CODE REVIEW

File: connector_builder_mcp/_connector_builder.py [hunk 1]
```diff
@@ -146,6 +149,12 @@ def execute_stream_test_read(
         int,
         Field(description="Maximum number of records to read", ge=1, le=1000),
     ] = 10,
+    summary_only: Annotated[
```
Comment (aaronsteers) [inline connector_builder_mcp/_connector_builder.py:152]: Default to False explicitly if not set

File: connector_builder_mcp/_connector_builder.py [hunk 2]
```diff
@@ -154,9 +163,10 @@ def execute_stream_test_read(
         config: Connector configuration
         stream_name: Name of the stream to test
         max_records: Maximum number of records to read
+        summary_only: If True, return only summary information. If False, include actual record data.
 
     Returns:
-        Test result with success status and details
+        Test result with success status and details, optionally including record data
```
Comment (aaronsteers) [inline connector_builder_mcp/_connector_builder.py:169]: Toggle this to: optionally *excluding* data

File: connector_builder_mcp/_connector_builder.py [hunk 3]
```diff
@@ -185,22 +207,79 @@ def execute_stream_test_read(
...
+        )
 
     except Exception as e:
         logger.error(f"Error testing stream read: {e}")
         error_msg = f"Stream test error: {str(e)}"
-        return StreamTestResult(success=False, message=error_msg, errors=[error_msg])
+
+        slices_data = None
+
+        if include_raw_response_data:
```
Comment (aaronsteers) [inline connector_builder_mcp/_connector_builder.py:268]:
```suggestion
        if include_raw_response_data is not False:
```

File: connector_builder_mcp/_connector_builder.py [hunk 4]
```diff
@@ -146,6 +152,16 @@ def execute_stream_test_read(
...
+    include_records: Annotated[
+        bool,
+        Field(description="If True, include actual record data in the response"),
+    ] = False,
+    include_raw_response_data: Annotated[
+        bool,
+        Field(
+            description="If True, include raw HTTP request and response data in slices structure"
+        ),
+    ] = False,
```
Comment (aaronsteers) [inline connector_builder_mcp/_connector_builder.py:164]:
```suggestion
    include_raw_response_data: Annotated[
        bool | None,
        Field(
            description="If True, include raw HTTP request and response data in slices structure"
        ),
    ] = None,
```

File: connector_builder_mcp/_connector_builder.py [hunk 5]
```diff
@@ -185,22 +207,79 @@ def execute_stream_test_read(
...
+
+            if result.record and result.record.data:
+                try:
+                    stream_data = result.record.data
+
+                    if include_records:
+                        if isinstance(stream_data, dict) and "records" in stream_data:
+                            records_data = stream_data["records"]
+                        elif isinstance(stream_data, dict):
+                            records_data = [stream_data]
```
Comment (aaronsteers) [inline connector_builder_mcp/_connector_builder.py:221]: Why do we have this conditional? Is record data sometimes under ["records"] and sometimes not? This needs an explanation if so. Otherwise, feels like one of these paths is dead code.

File: connector_builder_mcp/_connector_builder.py [hunk 6]
```diff
@@ -46,6 +46,12 @@ class StreamTestResult(BaseModel):
     message: str
     records_read: int = 0
     errors: list[str] = []
+    records: list[dict[str, Any]] | None = Field(
+        default=None, description="Actual record data when include_records=True"
+    )
+    slices: list[dict[str, Any]] | None = Field(
+        default=None, description="Raw slices data when include_raw_response_data=True"
```
Comment (aaronsteers) [inline connector_builder_mcp/_connector_builder.py:53]: Use my description. You'll need to format though for line length. Pull before updating.

```suggestion
        default=None, description="By default, raw request and response data will be returned only upon error. Set to True or False, to always or never return the raw response data."
```

File: connector_builder_mcp/_connector_builder.py [hunk 7]
```diff
@@ -154,9 +171,16 @@ def execute_stream_test_read(
         config: Connector configuration
         stream_name: Name of the stream to test
         max_records: Maximum number of records to read
+        include_records: If True, include actual record data in the response
```
Comment (aaronsteers) [inline connector_builder_mcp/_connector_builder.py:174]: Remove this entire args block. It is redundant with the descriptions we're putting inline above.

File: connector_builder_mcp/_connector_builder.py [hunk 8]
```diff
@@ -154,9 +173,16 @@ def execute_stream_test_read(
...
+        include_raw_response_data: If True, include raw HTTP request and response data in slices structure.
```
Comment (aaronsteers) [inline connector_builder_mcp/_connector_builder.py]: My previous comment was intended to update the help text for the 'include_raw_response_data' arg:

>By default, raw request and response data will be returned only upon error. Set to True or False, to always or never return the raw response data.

I want this to be what the LLM sees as guidance.
